x-pack/protocol/src/test/java/org/elasticsearch/protocol/xpack/ml/job/config/AnalysisConfigTests.java

```diff
@@ -0,0 +1,285 @@
+                            Map<String, Object> charFilter = new HashMap<>();
+                            charFilter.put("type", "pattern_replace");
+                            charFilter.put("pattern", pattern);
+                            analyzerBuilder.addCharFilter(charFilter);
+                        }
+                    }
+
+                    Map<String, Object> tokenizer = new HashMap<>();
+                    tokenizer.put("type", "pattern");
+                    tokenizer.put("pattern", randomAlphaOfLength(10));
+                    analyzerBuilder.setTokenizer(tokenizer);
+
+                    if (randomBoolean()) {
+                        for (String pattern : generateRandomStringArray(4, 40, false)) {
+                            Map<String, Object> tokenFilter = new HashMap<>();
+                            tokenFilter.put("type", "pattern_replace");
+                            tokenFilter.put("pattern", pattern);
+                            analyzerBuilder.addTokenFilter(tokenFilter);
+                        }
+                    }
+                }
+                builder.setCategorizationAnalyzerConfig(analyzerBuilder.build());
+            }
+        }
+        if (randomBoolean()) {
+            builder.setLatency(TimeValue.timeValueSeconds(randomIntBetween(1, 1_000_000)));
+        }
+        if (randomBoolean()) {
+            builder.setMultivariateByFields(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.setOverlappingBuckets(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.setResultFinalizationWindow(randomNonNegativeLong());
+        }
+
+        boolean usePerPartitionNormalisation = randomBoolean();
+        builder.setUsePerPartitionNormalization(usePerPartitionNormalisation);
+        if (!usePerPartitionNormalisation) { // influencers can't be used with per partition normalisation
+            builder.setInfluencers(Arrays.asList(generateRandomStringArray(10, 10, false)));
+        }
+        return builder;
+    }
+
+    @Override
+    protected AnalysisConfig createTestInstance() {
+        return createRandomized().build();
+    }
+
+    @Override
+    protected AnalysisConfig doParseInstance(XContentParser parser) {
+        return AnalysisConfig.PARSER.apply(parser, null).build();
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
+    }
+
+    public void testBuild_GivenMlCategoryUsedAsByFieldAndCategorizationFieldName() {
+        Detector.Builder detector = new Detector.Builder();
+        detector.setFunction("count");
+        detector.setOverFieldName("mlcategory");
+        AnalysisConfig.Builder ac = new AnalysisConfig.Builder(Collections.singletonList(detector.build()));
+        ac.setCategorizationFieldName("msg");
+        ac.build();
+    }
+
+    public void testBuild_GivenNonOverlappingNestedFields() {
```

Comment:
This was testing the complex validation in the builder that's been removed, so it's not worth having in the client.